### ISSUE TITLE
I wrote an automatic Index.md generator for smeagol-wiki

### ISSUE THREAD

**Issue by jrstrick** [AustinWise/smeagol#57]
I switched to Smeagol-Wiki from deadwiki, which generated default indexes where there weren't any. So to avoid having to write dozens of Index.md files for a very dynamic wiki (I use it as a personal KMS), I wrote automation for the job. Available here: [https://github.com/jrstrick/Make_Index-for-Smeagol-Wiki](https://github.com/jrstrick/Make_Index-for-Smeagol-Wiki).

Thanks for making smeagol-wiki. It's been a godsend.

**Comment by AustinWise** [comment]
That's a handy tool. One could even use it with tools besides Smeagol, such as in GitHub repos.

It does make me think that there are two things Smeagol could do better:
* When there is no `index.md` file, show a list of files.
* Make the "overview" button show files in the current folder, not the repository root.

**Comment by AustinWise** [comment]
Thanks for using Smeagol and then for the ideas on how to make it better! Version 0.4.4 is now available with the two enhancements listed above.

**Comment by jrstrick** [comment]
Awesome! I'll give it a whirl.

**Comment by jrstrick** [comment]
When I wrote make_index, the idea that it might be useful for non-smeagol applications hadn't occurred to me. Now that you've mentioned it, I've updated make_index to let the user select the index filename. It defaults to README.md, to better sync with smeagol. It will also read a smeagol.toml file, if present, and use that index filename.
If you're still interested in adding features to smeagol, I'd like to request two:

1. A "New" button, to create new markdown files in the current directory. Bonus points if it lets me pull in a template.
2. A "markdown_editor" field in smeagol.toml that lets me open external editors when the new or edit buttons are clicked.

*or*

3. The ability to add buttons that call external programs in the smeagol.toml file generally. This would allow me to kick make_index off from my web browser *and* add a new button and an edit in external editor button, and all kinds of other things.

If you don't have the time to investigate this, I'll see what I can do, but I am a *complete* rust newb.
-JRS
